### ISSUE TITLE
Allow analysis start/stop and visualisation of analysis progress

### ISSUE THREAD

**Issue by sksmi** [SACGF/variantgrid#222]
**Description**
VG analysis frequently errors out when developing/editing an analysis workflow as it will be loading nodes while I'm moving nodes around or setting node filters. Can be frustratingly slow when trying to set up as you can't change the filters until the node has loaded, but you know you don't want see that data anyway.  Would be ideal if you could add a bunch of nodes and/or set the filters then press something to start analysis. 

Would also be great to be able to see the status of an analysis & cancel an analysis - I know I've caused lots of problems by kicking of multiple analyses at once then trying to delete a bunch (for valid reasons, and often bca I don't think anything is happening).

e.g. this happens to me most times I'm developing an analysis - once I'm done setting it up I always force re-load (either because I have to, or just to make sure that it's loaded correctly).
![image.png](https://images.zenhubusercontent.com/5bc5465086f3a678d497c0be/8ece8818-c8e6-4c42-9a29-9fd95ef74689)

**Additional context**
Think this is a fairly widespread issue for anyone who is not using a template.

**Comment by davmlaw** [comment]
Always being able to see node config is covered in issue #51 

But yeah separating the construction vs execution would make it way quicker to build things.

I think this should be an option - maybe have a play/pause button in the analysis toolbox for auto-execute/load nodes or not. Then have initial setting from user settings

**Comment by davmlaw** [comment]
Having a count of active jobs / total jobs in the top toolbar would also be nice - like the celery monitoring page but in an analysis

Suggestion from another issue is to "move the force reload button to the toolbar"

If people wanted to do this, they could leave a sample unselected in a source node, build things, then select the sample at the end. The fact that they don't do that, to me, shows their revealed preferences of how they want things to work.

**Comment by sksmi** [comment]
> If people wanted to do this, they could leave a sample unselected in a source node, build things, then select the sample at the end. The fact that they don't do that, to me, shows their revealed preferences of how they want things to work.

Not sure about this one - as a user I didn't know this was possible so I'm guessing most other people wouldn't. Also a bunch of reasons why you wouldn't do this unless specifically instructed - 1) in the common case you'd load from the template on the sample page then modify so you've already got the sample loading, 2) the gene list node doesn't load unless there's a sample, 3) it's counterintuitive - users tend to work from top to bottom solving the errors as they go, so they're unlikely to not load the sample as the first point. 4) Modifying an analysis is usually multi-step e.g. add a couple of nodes, load, assess the result, modify.. load.. assess etc. I think the most common use case would be when adding gene/phenotype nodes.

**Comment by davmlaw** [comment]
🤖 Written by Claude

## Current state (Sept 2026)

Most of this issue has been addressed by other work:

- **Editor doesn't wait for the grid** - #51. Once a node is loaded, `node_data_grid.html` loads the editor straight away and fetches the grid rows separately.
- **Large grids don't auto-load** - #51. Nodes at or above `ANALYSIS_NODE_GRID_AUTO_LOAD_MAX_VARIANTS` (default 50k, adjustable in user/lab/org settings) show a "load variants" placeholder instead of running the row query.
- **Analysis progress** - the analysis toolbar shows active/queued task counts.
- **Cancel** - a loading node can be cancelled (`node_cancel_load`), which also cancels its SQL.
- **Errors while editing/moving nodes** - node loading is now lease-based scheduling rather than prebuilt task chains. A load overtaken by an edit exits quietly instead of erroring.

## Remaining gap

**The editor isn't shown while a node is loading.** Clicking a node that is dirty, queued or loading goes to `node_async_wait.html`, which shows the spinner and a Cancel button but never loads the editor. Editing a node puts it and all its descendants back into loading, so you still can't change a descendant's filters until its load finishes - the original complaint here. #51 fixed waiting on the grid, not waiting on the node load.

## Proposed next steps

1. Load the editor on the async wait page as well, with the same stale-response guard the grid page uses. First check which editors read `node.count` or the queryset when rendering, and make them cope with a node that has no counts yet.
2. When an edit bumps node versions, cancel the in-flight loads it overtakes (the `NodeTask` already stores the DB pid). Currently their SQL runs to completion before the load notices it's out of date, so rapid edits pile up worker time.
3. Hold off on a play/pause "auto-execute" toggle. With 1 and 2, an unwanted load costs worker time rather than blocking the user, while a pause adds a new "dirty but held" state for the scheduler, backlog sweep and template runs. Revisit if node load time itself (see `NodeVersion.load_data["timings"]`) turns out to be the problem.
